src/os/bluestore/BlueStore.cc

```diff
@@ -8014,8 +8014,10 @@ int BlueStore::_fsck_on_open(BlueStore::FSCKDepth depth, bool repair)
   _fsck_collections(&errors);
   used_blocks.resize(fm->get_alloc_units());
 
-  int r = bluefs->get_block_extents(bluefs_layout.shared_bdev, &bluefs_extents);
-  ceph_assert(r == 0);
+  if (bluefs) {
+    int r = bluefs->get_block_extents(bluefs_layout.shared_bdev, &bluefs_extents);
```

Comment:
@hualongfeng could you also move `interval_set<uint64_t> bluefs_extents;` and the `for` iterating over `bluefs_extents`  into this `if` block as well?

also, i think we could be more specific in the title:

>  fixed a bug when unusing bluefs

how about:

> do not update used_blocks with bluefs_extents if bluestore is not available